### ISSUE TITLE
Allow restriction of API access to logged-in users

### ISSUE THREAD

**Issue by ajdlinux** [getpatchwork/patchwork#325]
Requested by Konstantin: https://lists.ozlabs.org/pipermail/patchwork/2019-November/006369.html

Site admins may want to restrict API access to authenticated users only to control server load.

**Comment by daxtens** [comment]
Do we want to go the whole hog and implement proper ratelimiting, with
configurable limits for anonymous/logged in users? I'm thinking about how
to protect against misbehaving but authorised scripts...

**Comment by ajdlinux** [comment]
Yeah why not, DRF seems to support that in a very straightforward manner

**Comment by ajdlinux** [comment]
Happy to take this, feel free to assign to me

**Comment by ajdlinux** [comment]
Having asked on IRC, I was helpfully pointed to https://www.django-rest-framework.org/api-guide/permissions/#setting-the-permission-policy

```
REST_FRAMEWORK = {
    'DEFAULT_PERMISSION_CLASSES': [
        'rest_framework.permissions.IsAuthenticated',
    ]
}
```

seems like it'd probably do it.

DRF also has throttling options: https://www.django-rest-framework.org/api-guide/throttling/

@mricon would overriding the built-in DRF permission and throttling settings work for your use case?

**Comment by ajdlinux** [comment]
Ugh turns out we define our own permissions class and use that, but we can easily fix that class
